websockets/websockets_test.go

```diff
@@ -721,3 +725,50 @@ func TestOncloseDefineWithInvalidValue(t *testing.T) {
 	assert.Error(t, err)
 	assert.Contains(t, err.Error(), "a value for 'onclose' should be callable")
 }
+
+func TestCustomHeaders(t *testing.T) {
+	t.Parallel()
+	ts := newTestState(t)
+	sr := ts.tb.Replacer.Replace
+
+	mu := &sync.Mutex{}
+	collected := make(http.Header)
+
+	ts.tb.Mux.HandleFunc("/ws-echo-someheader", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		responseHeaders := w.Header().Clone()
+		conn, err := (&websocket.Upgrader{}).Upgrade(w, req, responseHeaders)
+		if err != nil {
+			t.Fatalf("/ws-echo-someheader cannot upgrade request: %v", err)
+		}
+
+		mu.Lock()
+		collected = req.Header.Clone()
+		mu.Unlock()
+
+		err = conn.Close()
+		if err != nil {
+			t.Logf("error while closing connection in /ws-echo-someheader: %v", err)
+		}
+	}))
+
+	ts.state.CookieJar, _ = cookiejar.New(nil)
+	ts.ev.WaitOnRegistered()
+	err := ts.ev.Start(func() error {
+		_, runErr := ts.rt.RunString(sr(`
+		var ws = new WebSocket("WSBIN_URL/ws-echo-someheader", [], {headers: {"x-lorem": "ipsum"}})
```

Comment:
```suggestion
		var ws = new WebSocket("WSBIN_URL/ws-echo-someheader", null, {headers: {"x-lorem": "ipsum"}})
```

does this work?

Reply:
it should, since for now now restrictions for the second arg, we don't use it